File: uniq_tool.py
from pathlib import Path
from check_file_permission import check_file_permission

def uniq_file(file_path, config):
    if not check_file_permission(file_path, "write", config):
        raise PermissionError(f"Write access denied for {file_path}")
    file_path = Path(file_path).resolve()
    if not file_path.is_file():
        raise ValueError(f"File not found: {file_path}")
    with open(file_path, "r", encoding="utf-8") as f:
        lines = f.readlines()
    unique_lines = list(dict.fromkeys(lines))
    with open(file_path, "w", encoding="utf-8") as f:
        f.writelines(unique_lines)
    return f"Removed duplicates from {file_path}"

File: ts-tools/uniq_tool.ts


import * as fs from 'fs';
import * as path from 'path';
import { checkFilePermission } from './check_file_permission';

export function uniqFile(filePath: string, config: any): Promise<string> {
    return new Promise((resolve, reject) => {
        checkFilePermission(filePath, "write", config)
            .then(() => {
                const resolvedPath = path.resolve(filePath);
                fs.readFile(resolvedPath, 'utf-8', (err, data) => {
                    if (err) {
                        reject(err);
                    } else {
                        const lines = data.split('\n');
                        const uniqueLines = Array.from(new Set(lines));
                        fs.writeFile(resolvedPath, uniqueLines.join('\n'), 'utf-8', (err) => {
                            if (err) {
                                reject(err);
                            } else {
                                resolve(`Removed duplicates from ${filePath}`);
                            }
                        });
                    }
                });
            })
            .catch(reject);
    });
}

